### PR TITLE
Benchmarks for identity pallet

### PR DESCRIPTION
**Pull request by maxsam4**
- Benchmarking harness upgraded to substrate 2.0.0 syntax.
- Benchmarks for the Identity pallet have been added and the pallet has been modified to use `WeightInfo` trait so that the weight info can be fed via auto-generated files.
- `set_primary_key` has been removed since it is not used anymore. To change the primary key of an identity, the current primary key must create and authorization that needs to be accepted by the new primary key (and maybe a CDD provider).
- `im-online` benchmarks have been disabled for now since they were broken.
- `ExtrinsicBaseWeight` has been increased from 0.5ms to 0.8ms to align better with our target hardware.

Note: Weight values being used right now are based on my local machine. There were other tasks running in the background while the benchmarks were running so these are not reliable but just a placeholder. We'll run final benchmarks on standardized hardware.

### CODE REVIEW

File: pallets/identity/src/lib.rs [hunk 1]
```diff
@@ -465,43 +455,16 @@ decl_module! {
             Ok(())
         }
 
-        /// Sets a new primary key for a DID.
-        ///
-        /// # Failure
-        /// Only called by primary key owner.
-        #[weight = 800_000_000]
-        fn set_primary_key(origin, new_key: T::AccountId) -> DispatchResult {
```
[the remark this answers is not in the document]
Comment (maxsam4) [reply pallets/identity/src/lib.rs:473]: This should have been removed a while back. We no longer allow users to change the primary key unilaterally.

The new primary key must accept the change and optionally a CDD provider as well. i.e. https://github.com/PolymathNetwork/Polymesh/blob/7837422c498f682dfed70eb989ab8abe36a57ca7/pallets/identity/src/lib.rs#L502 should be used instead.

File: primitives/src/ticker.rs [hunk 2]
```diff
@@ -104,6 +104,12 @@ impl Ticker {
     pub fn iter(&self) -> sp_std::slice::Iter<'_, u8> {
         self.0.iter()
     }
+
+    /// Returns a ticker that may not be in upper case
+    #[inline]
+    pub fn unsafe_ticker(raw: [u8; TICKER_LEN]) -> Ticker {
+        Ticker(raw)
```
[the remark this answers is not in the document]
Comment (maxsam4) [reply primitives/src/ticker.rs:111]: Actually, we don't need this function. I added this to test something but then forgot to remove it. Will remove now, thanks for catching!

File: pallets/identity/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Centril) [reply pallets/identity/src/lib.rs:473]: Ah; so its a drive-by change? Possibly extract to a different PR or add to PR description?

[the remark this answers is not in the document]
Comment (maxsam4) [reply pallets/identity/src/lib.rs:473]: Yeah, I should have done that. There are a couple more drive-by changes in this PR. I'll update the description.

**Review by adamdossa — APPROVED** [review]

**Review by vkomenda — APPROVED** [review]

**Review by satyamakgec — APPROVED** [review]

**Pull request merged**